### ISSUE TITLE
torch.compile loses channels_last memory format for reflect F.pad

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#183770]
### 🐛 Describe the bug

`torch.compile` with the Inductor backend does not preserve the output memory format for `F.pad(..., mode="reflect")`.

For a channels-last input tensor, eager execution returns a channels-last output, while the compiled version returns a contiguous NCHW-layout output. The values are equal, but the output strides and memory format differ.

```
import torch
import torch.nn.functional as F

def f(x):
    return F.pad(x, (2, 1, 1, 2), mode="reflect")

x = torch.randn(2, 3, 5, 4).to(memory_format=torch.channels_last)

eager = f(x.clone())
compiled = torch.compile(f, backend="inductor")(x.clone())

print("input_stride:", x.stride())
print("eager_stride:", eager.stride())
print("compiled_stride:", compiled.stride())
print("eager_channels_last:", eager.is_contiguous(memory_format=torch.channels_last))
print("compiled_channels_last:", compiled.is_contiguous(memory_format=torch.channels_last))
print("values_equal:", torch.equal(eager, compiled))

if (
    torch.equal(eager, compiled)
    and eager.is_contiguous(memory_format=torch.channels_last)
    and not compiled.is_contiguous(memory_format=torch.channels_last)
):
    raise SystemExit(0)

raise SystemExit(1)
```

### Error logs

```
input_stride: (60, 1, 12, 3)
eager_stride: (168, 1, 21, 3)
compiled_stride: (168, 56, 7, 1)
eager_channels_last: True
compiled_channels_last: False
values_equal: True
```

### Versions

PyTorch version: 2.13.0.dev20260513+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.11.15 (main, Mar 11 2026, 17:20:07) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-20-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA RTX 6000 Ada Generation
GPU 1: NVIDIA RTX 6000 Ada Generation

Nvidia driver version: 570.211.01
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

Versions of relevant libraries:
[pip3] numpy==2.4.4
[pip3] torch==2.13.0.dev20260513+cpu
[conda] numpy                      2.4.4                   pypi_0           pypi
[conda] torch                      2.13.0.dev20260513+cpu  pypi_0           pypi

cc @jamesr66a @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by karthickai** [comment]
**🤖 PT2 Repro Agent**

I tried to reproduce this issue on PyTorch nightly (`2.13.0.dev20260514+cu130`) on NVIDIA H100 and confirmed it reproduces (3/3 runs failed).

All 3 runs reproduced the reported behavior identically. With channels_last input (stride (60, 1, 12, 3)), eager F.pad(..., mode="reflect") returns a channels_last output (stride (168, 1, 21, 3), is_contiguous(channels_last)=True), while torch.compile(backend="inductor") returns an NCHW-contiguous output (stride (168, 56, 7, 1), is_contiguous(channels_last)=False). values_equal=True across all runs. test_fn exited 1 on all 3 runs because eager and compiled disagree on memory format.

<details>
<summary>Error output</summary>

```
input_stride: (60, 1, 12, 3)
eager_stride: (168, 1, 21, 3)
compiled_stride: (168, 56, 7, 1)
eager_channels_last: True
compiled_channels_last: False
values_equal: True

ERROR conda.cli.main_run:execute(49): `conda run python repro.py` failed. (See above for error)
```
</details>

<details>
<summary>Repro script I used</summary>

```python
import os
import torch
import torch.nn.functional as F


def f(x):
    return F.pad(x, (2, 1, 1, 2), mode="reflect")


def test_fn() -> bool:
    """Return True if NO bug. False if compiled output loses channels_last.

    The reported bug: eager output preserves channels_last memory format but
    inductor-compiled output returns NCHW contiguous (values equal but strides differ).
    """
    torch._dynamo.reset()
    backend = os.environ.get("TORCH_COMPILE_BACKEND", "inductor")

    x = torch.randn(2, 3, 5, 4).to(memory_format=torch.channels_last)

    eager = f(x.clone())
    compiled = torch.compile(f, backend=backend)(x.clone())

    values_equal = torch.equal(eager, compiled)
    eager_cl = eager.is_contiguous(memory_format=torch.channels_last)
    compiled_cl = compiled.is_contiguous(memory_format=torch.channels_last)

    # No bug only if values equal AND memory format preservation matches eager
    return values_equal and (eager_cl == compiled_cl)


if __name__ == "__main__":
    x = torch.randn(2, 3, 5, 4).to(memory_format=torch.channels_last)
    eager = f(x.clone())
    compiled = torch.compile(f, backend="inductor")(x.clone())

    print("input_stride:", x.stride())
    print("eager_stride:", eager.stride())
    print("compiled_stride:", compiled.stride())
    print("eager_channels_last:", eager.is_contiguous(memory_format=torch.channels_last))
    print("compiled_channels_last:", compiled.is_contiguous(memory_format=torch.channels_last))
    print("values_equal:", torch.equal(eager, compiled))

    try:
        ok = test_fn()
    except Exception:
        import traceback; traceback.print_exc()
        os._exit(2)
    os._exit(0 if ok else 1)

```
</details>

<details>
<summary>My environment</summary>

```
Collecting environment information...
PyTorch version: 2.13.0.dev20260514+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: CentOS Stream 9 (x86_64)
GCC version: (GCC) 11.5.0 20240719 (Red Hat 11.5.0-14)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.13.12 | packaged by Anaconda, Inc. | (main, Feb 24 2026, 16:13:31) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.16.1-0_fbk2_hardened_0_gf40efc324cc8-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA H100
Nvidia driver version: 580.126.09
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  46
On-line CPU(s) list:                     0-45
Vendor ID:                               AuthenticAMD
Model name:                              AMD EPYC 9654 96-Core Processor
CPU family:                              25
Model:                                   17
Thread(s) per core:                      1
Core(s) per socket:                      46
Socket(s):                               1
Stepping:                                1
BogoMIPS:                                4792.79
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw perfctr_core ssbd ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx512_bf16 clzero xsaveerptr wbnoinvd arat npt lbrv nrip_save tsc_scale vmcb_clean flushbyasid pausefilter pfthreshold v_vmsave_vmload vgif vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid fsrm flush_l1d arch_capabilities
Virtualization:                          AMD-V
Hypervisor vendor:                       KVM
Virtualization type:                     full
L1d cache:                               2.9 MiB (46 instances)
L1i cache:                               2.9 MiB (46 instances)
L2 cache:                                23 MiB (46 instances)
L3 cache:                                736 MiB (46 instances)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-45
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Vulnerable
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:                Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Not affected; BHI: Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Vulnerable
Vulnerability Tsx async abort:           Not affected

Versions of relevant libraries:
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.4.4
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] torch==2.13.0.dev20260514+cu130
[pip3] torchao==0.17.0
[pip3] torchaudio==2.11.0.dev20260515+cu130
[pip3] torchdata==0.11.0
[pip3] torchmetrics==1.0.3
[pip3] torchrec==1.7.0.dev20260413+cu130
[pip3] torchtune==0.6.1
[pip3] torchvision==0.28.0.dev20260515+cu130
[pip3] triton==3.7.0+git88b227e2
[conda] Could not collect
```
</details>

---
If the repro agent produced an incorrect result (wrong classification, bad repro script, missed the actual bug, etc.), add the label `repro:agent:h100:incorrect` to the issue.

**Comment by morrison-turnansky** [comment]
working on this

**Comment by morrison-turnansky** [comment]
although this was originally reproducible, it looks like change to fix this in my pr was merged via another recent commit to main. I have closed my pr. issue is fixed.

**Comment by jansel** [comment]
My agent was unable to reproduce this on 8dd44388526b4c638900a96f16ecfd1c0b2b0f8d. Reopen if you see this on the latest PyTorch nightly
